### ISSUE TITLE
Language redirection with or without URL prefix does not work, no index alias assigned

### ISSUE THREAD

**Issue by Ainschy** [contao/contao#6328]
### Affected version(s)

4.13.29

### Description

In the installation there are two page trees one for NL without URL prefix and one with EN with URL prefix EN. In the page trees there is no page with the alias index.

Language forwarding is disabled for NL, language fallback is enabled for EN.

No matter with which language I call the domain.tld, I always land on the first page of the NL page tree. 
Only when the URL prefix is set at NL, a correct redirection takes place. The language redirection option has no effect.

Legacy routing is disabled

![2023-08-23_12h33_25](https://github.com/contao/contao/assets/1788135/67f08c88-e2b1-4676-984d-a4d825c78360)
![2023-08-23_12h33_49](https://github.com/contao/contao/assets/1788135/0f946b87-9773-40d0-a308-6057b748c59d)

**Comment by ausi** [comment]
As a side note: if the non-prefixed root page **does** have a page with alias `index` this behaviour would be correct. (A potential fix for this issue should keep that in mind.)

> No matter with which language I call the domain.tld, I always land on the first page of the NL page tree.

Does the page just show up, or do you get redirected to the URL of the NL home page?

**Comment by fritzmg** [comment]
> Does the page just show up, or do you get redirected to the URL of the NL home page?

It's a 302 redirect to the NL home page (checked via the domain provided in the Slack thread).

**Comment by Ainschy** [comment]
The behavior can be reproduced in the contao demo.

**Comment by aschempp** [comment]
I did have a look at the code base, and the `disableLanguageRedirect` checkbox _currently_ only has an effect if you **do** have a URL prefix. It does nothing if the URL prefix is empty.

To sum up what I reproduced:
 - Page tree with language `en` and URL prefix `en`, which is the fallback language
 - Page tree with language `nl` and empty URL prefix. `disableLanguageRedirect` checkbox is set
 - Both page trees have a `home` aliased page, no `index` page
 - I'm being redirected to the `/home` page in `nl`

@contao/developers If we agree we would expect to be redirected to the english home page (due to `disableLanguageRedirect` being set), I can provide an easy fix 😊
